### PR TITLE
Schedule added to context for all orchestrators

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Schedule added to context for all orchestrators, implemented warning for local orchestrator as it is not implemented there. Currently as darft waiting for ENG-58 to be merged and remains to be fully tested

### CODE REVIEW

File: src/zenml/orchestrators/local/local_orchestrator.py [hunk 1]
```diff
@@ -75,6 +76,11 @@ def run_pipeline(
         if runtime_configuration is None:
             runtime_configuration = RuntimeConfiguration()
 
+        if runtime_configuration.schedule:
+            logger.warning("Local Orchestrator currently does not support the"
```
Comment (schustmi) [inline src/zenml/orchestrators/local/local_orchestrator.py:80]: This same warning should also happen for the `AirflowOrchestrator` right?

File: src/zenml/orchestrators/context_utils.py [hunk 2]
```diff
@@ -49,3 +51,62 @@ def add_stack_as_metadata_context(
     for k, v in stack_dict.items():
         c_property = context.properties[k]  # type:ignore[attr-defined]
         c_property.field_value.string_value = v
+
+
+def add_schedule_as_metadata_context(
```
Comment (schustmi) [inline src/zenml/orchestrators/context_utils.py:56]: I couldn't find any usages of this function, is `add_pydantic_object_as_metadata_context` the generic version for it and we can get rid of it entirely?

File: src/zenml/orchestrators/context_utils.py [hunk 3]
```diff
@@ -49,3 +51,62 @@ def add_stack_as_metadata_context(
...
+        context: a context proto message within a pipeline node
+    """
+    context.type.name = obj.__repr_name__().lower()
+    # Setting the name of the context
+    name = str(hash(obj.json(sort_keys=True)))
+    context.name.field_value.string_value = name  # type:ignore[attr-defined]
+
+    # Setting the properties of the context
+    for k, v in obj.dict().items():
+        c_property = context.properties[k]  # type:ignore[attr-defined]
```
Comment (schustmi) [inline src/zenml/orchestrators/context_utils.py:104]: More of a question than a request: For all the execution properties of a step, we only use the `string_value` of the property and always do a `json.dumps(value)` to serialize the values to a string. Do you see any pros/cons to both solutions and should we maybe try to unify them? (Probably also @bcdurak, he's probably the expert here)

**Review by schustmi — CHANGES_REQUESTED** [review]

File: src/zenml/orchestrators/local/local_orchestrator.py (hunk 1 above)
Comment (AlexejPenner) [reply src/zenml/orchestrators/local/local_orchestrator.py:80]: Implemented true scheduling for airflow now - non-schduled airflow runs will run once immediately

File: src/zenml/orchestrators/context_utils.py (hunk 2 above)
Comment (AlexejPenner) [reply src/zenml/orchestrators/context_utils.py]: I removed it - it was an artifact of development

File: src/zenml/orchestrators/context_utils.py (hunk 3 above)
Comment (AlexejPenner) [reply src/zenml/orchestrators/context_utils.py:75]: Not sure I fully understand the question. @bcdurak you got anything?

File: src/zenml/integrations/kubeflow/orchestrators/kubeflow_dag_runner.py [hunk 4]
```diff
@@ -390,12 +407,18 @@ def _generate_tfx_ir(
...
+        self,
+        pipeline: "BasePipeline",
+        stack: "Stack",
+        runtime_configuration: "RuntimeConfiguration",
+    ) -> None:
         """Compiles and outputs a Kubeflow Pipeline YAML definition file.
         Args:
           pipeline: The logical TFX pipeline to use when building the Kubeflow
             pipeline.
           stack: The ZenML stack that the pipeline is running on
```
Comment (htahir1) [inline src/zenml/integrations/kubeflow/orchestrators/kubeflow_dag_runner.py:420]: Minor nits: Maybe full stops both these?

File: src/zenml/orchestrators/context_utils.py [hunk 5]
```diff
@@ -49,3 +51,33 @@ def add_stack_as_metadata_context(
...
+
+    Args:
+        obj: an instance of a pydantic object
+        context: a context proto message within a pipeline node
+    """
+    context.type.name = (  # type: ignore[attr-defined]
+        obj.__repr_name__().lower()
+    )
+    # Setting the name of the context
+    name = str(hash(obj.json(sort_keys=True)))
```
Comment (htahir1) [inline src/zenml/orchestrators/context_utils.py:70]: How about a try catch for this if obj.json() fails? Or even the hash -> Although technically that should not be possible

File: src/zenml/orchestrators/context_utils.py (hunk 3 above)
Comment (htahir1) [reply src/zenml/orchestrators/context_utils.py:75]: I really think we should unify too, but something we can do later

**Review by htahir1 — COMMENTED** [review]
Minor nit picks, other than that its a go!

**Pull request merged**